connection.go

```diff
@@ -243,6 +251,25 @@ func (ch *Channel) newOutboundConnection(timeout time.Duration, hostPort string,
 		return nil, err
 	}
 
+	opts := &ch.connectionOptions
+	tosPriority := opts.TosPriority
+
+	if tosPriority > 0 {
+		// Verify we are using a TCP socket
```

Comment:
Can we move this into a separate function that takes the connection + `tosPriority`?